internal/loader/synthetic_device_loader.go

```diff
@@ -55,3 +65,23 @@ func (sd *SyntheticDeviceLoader) BatchGetSyntheticDeviceByVehicleID(ctx context.
 
 	return results
 }
+
+// BatchGetSyntheticDeviceByVehicleID implements the dataloader for finding synthetic devices by their ids and returns
+// them in the order requested
+func (sd *SyntheticDeviceLoader) BatchGetSyntheticDeviceByID(ctx context.Context, syntheticDeviceIDs []int) []*dataloader.Result[*gmodel.SyntheticDevice] {
+	results := make([]*dataloader.Result[*gmodel.SyntheticDevice], len(syntheticDeviceIDs))
+
+	devices, err := models.SyntheticDevices(models.SyntheticDeviceWhere.ID.IN(syntheticDeviceIDs)).All(ctx, sd.db.DBS().Reader)
+	if err != nil {
+		for i := range results {
+			results[i] = &dataloader.Result[*model.SyntheticDevice]{Error: err}
+		}
+		return results
+	}
+
+	for i, sdv := range devices {
+		results[i] = &dataloader.Result[*gmodel.SyntheticDevice]{Data: repositories.SyntheticDeviceToAPI(sdv)}
+	}
```

Comment:
I think you still have to build up a map and be careful about order. You do not know whether

* `syntheticDeviceIDs` is without repeats
* the items are coming back in the order that you want